src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/ReadMetadata.java

```diff
@@ -225,26 +245,38 @@ public float getZishScore( final String readGroup, final int fragmentSize ) {
         return getFragmentLengthStatistics(readGroup).getZishScore(fragmentSize);
     }
 
-    public int getGroupMedianFragmentSize( final String readGroup ) {
-        return getFragmentLengthStatistics(readGroup).getMedian();
-    }
-
     @VisibleForTesting
     LibraryStatistics getFragmentLengthStatistics( final String readGroup ) {
         return libraryToFragmentStatistics.get(getLibraryName(readGroup));
     }
 
     public long getNReads() { return nReads; }
+    public int getAvgReadLen() { return avgReadLen; }
     public long getNRefBases() { return nRefBases; }
     public int getNPartitions() { return partitionBounds.length; }
     public PartitionBounds getPartitionBounds( final int partitionIdx ) { return partitionBounds[partitionIdx]; }
     @VisibleForTesting PartitionBounds[] getAllPartitionBounds() { return partitionBounds; }
 
     public long getMaxReadsInPartition() { return maxReadsInPartition; }
-
-    public int getCoverage() {
+    public float getCoverage() {
         return coverage;
     }
+    public float getMeanBaseQuality() { return meanBaseQuality; }
+    public float getAccurateKmerCoverage( final int kSize ) {
+        // i.e., p = (1 - 10**(-Q/10))**K
+        final float probabilityOfAnAccurateKmer =
+                (float)Math.exp(kSize * Math.log1p(-Math.pow(10.,meanBaseQuality/-10.)));
+        return coverage * probabilityOfAnAccurateKmer;
+    }
+
+    public int getMeanPartitionSpan() {
```

Comment:
It looks like you're returning the median here (which is probably better) but maybe forgot to rename the method.

Reply:
There's `MathUtils.median()` that also handles cases like having an odd number of values properly in case you want to consider using that.

Reply:
I realize that efficiency doesn't much matter here -- it's a small array.  But I just can't bring myself to box a mess of ints into a Collection, convert that to an array of doubles, and do an elaborate sampling procedure when I can just sort a small array of primitives.
It's just me.